### DISCUSSION TITLE
Fail2ban Debian 12 not working

### DISCUSSION THREAD

**Discussion by Acenl12** [fail2ban/fail2ban#3677] [General]
I see a lot of connection attempts in my /var/log/auth.log, but those aren't getting banned by fail2ban on Debian 12. I tried moving the banaction to ipset but that didn't helped either.

Fail2ban version
1.0.2

It's an azure cloud vm and I had ssh wide open to the internet and literally no ip was banned
Based on my experience in the past there should be at least a couple banned

**Comment by sebres** [comment]
> But I do see a lot more connection attempts in my logs which should be banned but aren't

You didn't give the info what exactly should be banned... neither log-excerpt, nor what fail2ban see or not see.
I already provided you the link to [FAQ](https://github.com/fail2ban/fail2ban/wiki/How-fail2ban-works) - did you read it?
Seems not, because there is definitely described the difference between monitoring of journal vs. log-file (correct backend one needs to choose).

Also interesting is the output of fail2ban-regex (does it find matches?):
```bash
# for systemd journal:
?sudo? fail2ban-regex systemd-journal sshd
# for log-file:
?sudo? fail2ban-regex /var/log/auth.log sshd
```

> I see a lot of connection attempts in my /var/log/auth.log

Related to #3676 your `sshd` jail uses `backend` systemd - what means, it would monitor a systemd-journal, not a log-file (/var/log/auth.log), what is basically normal situation for Debian 12 (by default).

Anyway to switch to log-file monitoring, just add `backend = auto` to `[sshd]` section of `jail.local` (or where you enable it):
```ini
[sshd]
backend = auto
enabled = true
```
(and then restart fail2ban or sshd jail)

> I tried moving the banaction to ipset but that didn't helped either.

This has probably nothing to do with issue, if fail2ban (or rather filter of sshd jail) doesn't recognize failures at all.

**Comment by kwisatz** [comment]
I'll try to help @Acenl12 out here.

I was in the same boat with Debian 12 and fail2ban 1.0.2-2 which seems to be quite a mess on Debian 12. 
I should probably file a bug report over there or find the existing one and add my rant.

In any case, in Debian 12, the backend should be `systemd` for most jails, but the maintainer forgot to set a dependency on `python3-systemd`. 

You'll find an error such as `ERROR   Backend 'systemd' failed to initialize due to No module named 'systemd'` in your logs if this module is also missing for you.

**Reply by sebres** [reply]
Related to https://packages.debian.org/bookworm/fail2ban, it is there, just indeed as recommended only...
However related to [salsa.debian.org/.../fail2ban/47a2ffa9](https://salsa.debian.org/python-team/packages/fail2ban/-/commit/47a2ffa95ac787382f439505a257468ea666d17d) it is mandatory now (since 5 months).
As well as for trixie/sid it is the case (related to https://packages.debian.org/trixie/fail2ban) now...

@sylvestre, @fail2ban/maintainers: 
(It is probably better to add `python3-systemd` to mandatory dependencies for bookworm too, especially because by default sshd jail is enabled in debian package and its backend is probably by default systemd (I'm unsure it was ever fixed for Debian 12).
Could you check that, please...
